src/main/java/sg/dex/starfish/DataAsset.java

```diff
@@ -97,7 +103,7 @@ default String getContentHash() {
      * @return respective data asset sub class.
      * @throws UnsupportedOperationException if the Asset type is not DataAsset
      */
-    default DataAsset includeContentHash() {
+    default DataAsset includeContentHash() throws DigestException {
```

Comment:
Should this really be throwing a DigestException? Under what circumstances can this happen? Perhaps we should be catching and re-throwing if this happens?